packages/schemas/alterations/next-1718175155-custom-ui-assets.ts

```diff
@@ -0,0 +1,65 @@
+import { type CommonQueryMethods, sql } from '@silverhand/slonik';
+
+import type { AlterationScript } from '../lib/types/alteration.js';
+
+const getId = (value: string) => sql.identifier([value]);
+
+const getDatabaseName = async (pool: CommonQueryMethods) => {
+  const { currentDatabase } = await pool.one<{ currentDatabase: string }>(sql`
+    select current_database();
+  `);
+
+  return currentDatabase.replaceAll('-', '_');
+};
+
+const alteration: AlterationScript = {
+  up: async (pool) => {
+    const database = await getDatabaseName(pool);
+    const baseRoleId = getId(`logto_tenant_${database}`);
+
+    await pool.query(sql`
+      create table custom_ui_assets (
+        tenant_id varchar(21) not null
+          references tenants (id) on update cascade on delete cascade,
+        id varchar(21) not null,
+        files jsonb /* @use UploadFiles */ not null default '[]'::jsonb,
+        created_at timestamptz not null default(now()),
+        updated_at timestamptz not null default(now()),
+        primary key (id)
+      );
+      create index custom_ui_assets__id
+        on custom_ui_assets (tenant_id, id);
+
+      create trigger set_tenant_id before insert on custom_ui_assets
+        for each row execute procedure set_tenant_id();
+
+      alter table custom_ui_assets enable row level security;
+
+      create policy custom_ui_assets_tenant_id on custom_ui_assets
+        as restrictive
+        using (tenant_id = (select id from tenants where db_user = current_user));
+      create policy custom_ui_assets_modification on custom_ui_assets
+        using (true);
+
+      grant select, insert, update, delete on custom_ui_assets to ${baseRoleId};
+
+      alter table sign_in_experiences
+        add column custom_ui_assets_id varchar(21)
+        references custom_ui_assets (id) on update cascade on delete cascade;
+    `);
+  },
+  down: async (pool) => {
+    await pool.query(sql`
+      alter table sign_in_experiences
+        drop column custom_ui_assets_id;
+
+      drop policy custom_ui_assets_tenant_id on custom_ui_assets;
+      drop policy custom_ui_assets_modification on custom_ui_assets;
+      alter table custom_ui_assets disable row level security;
+      drop index custom_ui_assets__id;
```

Comment:
index, constrains and triggers will be automatically dropped when dropping a table.